lib/matplotlib/axes/_base.py

```diff
@@ -2450,29 +2450,33 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
             locator = axis.get_major_locator()
             x0, x1 = locator.nonsingular(x0, x1)
 
+            # Prevent margin addition from crossing a sticky value.  Small
+            # tolerances (whose value come from isclose()) must be used due to
+            # floating point issues with streamplot.
+            # Index of largest element < x0 + tol, if any.
+            i0 = stickies.searchsorted(x0 + 1e-5 * abs(x0) + 1e-8) - 1
```

Comment:
What is the `1e-8` doing for us here that the `1e-5*abs(x0)` is not?

Reply:
x0 could be equal to zero and we would still need some slop... the actual value comes from what isclose() uses, I didn't really want to bother finding something better.

Reply:
Can you factor that out into a function? This is used twice and the function removes the magic numbers.
~~~
def tol(x, rtol=1e-5, atol=1e-8):
    """A tolerance for x such that x+/-tol is considered close to x."""
    return rtol * abs(x) + atol

Reply:
That doesn't really work because in one case it's plus and in the other it's minus.  I moved the values to their own variables, though.

Reply:
But that would just be `x0 + tol(x0)` and `x1 - tol(x1)`.

Reply:
sure, pushed